controllers/lease_controller.go

```diff
@@ -0,0 +1,580 @@
+	if err != nil {
+		return nil, err
+	}
+	lc.ImageVector = imageVector
+	return lc, nil
+}
+
+// +kubebuilder:rbac:groups=druid.gardener.cloud,resources=etcds,verbs=get;list;watch
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;create;list;watch;update;patch;delete
+
+// Reconcile reconciles the compaction job.
+func (lc *LeaseController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	lc.logger.Info("Lease controller reconciliation started")
+	etcd := &druidv1alpha1.Etcd{}
+	if err := lc.Get(ctx, req.NamespacedName, etcd); err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return. Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{
+			RequeueAfter: 10 * time.Second,
+		}, err
+	}
+
+	if !etcd.DeletionTimestamp.IsZero() {
+		// Delete compaction job if exists
+		return lc.delete(ctx, lc.logger, etcd)
+	}
+
+	logger := lc.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String())
+
+	// Get delta snapshot lease to check the HolderIdentity value to take decision on compaction job
+	nsName := types.NamespacedName{
+		Name:      getDeltaLease(etcd),
+		Namespace: etcd.Namespace,
+	}
+
+	deltaLease := &coordinationv1.Lease{}
+	err := lc.Get(ctx, nsName, deltaLease)
+	if err != nil {
+		logger.Info("Couldn't fetch delta snap lease because: " + err.Error())
+
+		return ctrl.Result{
+			RequeueAfter: 10 * time.Second,
+		}, err
+	}
+
+	// Run compaction job
+	if etcd.Spec.Backup.Store != nil && lc.config.EnableCompactionJob {
+		hi, err := strconv.ParseInt(*deltaLease.Spec.HolderIdentity, 10, 64)
+		if err != nil {
+			logger.Error(err, "Can't convert holder identity of revision lease to integer")
+			return ctrl.Result{
+				RequeueAfter: 10 * time.Second,
+			}, err
+		}
+		// Reconcile job only when current revision is more than 1 million (this vale is configurable through `events-threshold` option in druid) than
+		// the last snapshot (full/compact) revision
+		if hi >= lc.config.EventsThreshold {
+			return lc.reconcileJob(ctx, logger, etcd)
+		}
+	}
+
+	return ctrl.Result{
+		Requeue: false,
+	}, nil
+}
+
+func (lc *LeaseController) reconcileJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
+	logger.Info("Reconcile etcd compaction job")
+
+	// Delete any existing cronjob if required.
+	// TODO(abdasgupta) : This is for backward compatibility towards ETCD-Druid 0.6.0. Remove it.
+	cronJob, err := lc.cleanCronJobs(ctx, logger, etcd)
+	if err != nil {
+		return ctrl.Result{
+			RequeueAfter: 10 * time.Second,
+		}, err
+	}
+
+	if cronJob != nil {
+		return ctrl.Result{Requeue: false}, nil
+	}
```

Comment:
See previous comment - I propose to delete `CronJobs` in `etcd_controller.go`

Reply:
Done